components/server/src/workspace/workspace-service.ts

```diff
@@ -151,6 +151,35 @@ export class WorkspaceService {
         return filtered;
     }
 
+    async listWorkspaces(
```

Comment:
it would be good if we start adding tests for new service methods, i think there is a framework setup already

Reply:
why cannot we use `getWorkspaces` instead?

Reply:
cc @svenefftinge for 👀

Reply:
Yes, we should use one method for this on the service layer as well. And it wouldn't hurt to rename it to `listWorkspaces` I believe

Reply:
> why cannot we use getWorkspaces instead?

@akosyakov  It's because I'd like to have only one API called one time in dashboard, that it list workspace with sort: `running` `pinned` `creationTime`. 

We call getWorkspaces two times in `/workspaces` page, which can't make pagination work with current UI, unless we move all pinned to another section: `recent` `recent pinned`. Or other UX change

Add new function will make db query easier than change and make it adapt both cases

Reply:
@akosyakov Since I changed to use one API call only, json-rpc implement part is very strange now 😞 

But if our plan is to design API but not create a server copy, I think it make sense to call only once with new API

Reply:
Do you mean drop JSON-RPC shim?

Reply:
>  Since I changed to use one API call only, json-rpc implement part is very strange now 😞

@mustard-mh we should not do it during migration, please see https://gitpod.slack.com/archives/C0463A2KQUU/p1699431787070909

Please also refrain refactoring services much it will slow us down.